src/wigner.jl

```diff
@@ -12,58 +12,114 @@ end
 WignerLaguerre(; parallel = false, tol = 1e-14) = WignerLaguerre(parallel, tol)
 
 @doc raw"""
-    wigner(state::QuantumObject, xvec::AbstractVector, yvec::AbstractVector; g::Real=√2,
-        solver::WignerSolver=WignerLaguerre())
-
-Generates the [Wigner quasipropability distribution](https://en.wikipedia.org/wiki/Wigner_quasiprobability_distribution)
-of `state` at points `xvec + 1im * yvec`. The `g` parameter is a scaling factor related to the value of ``\hbar`` in the
-commutation relation ``[x, y] = i \hbar`` via ``\hbar=2/g^2`` giving the default value ``\hbar=1``.
-
-The `solver` parameter can be either `WignerLaguerre()` or `WignerClenshaw()`. The former uses the Laguerre polynomial
-expansion of the Wigner function, while the latter uses the Clenshaw algorithm. The Laguerre expansion is faster for
-sparse matrices, while the Clenshaw algorithm is faster for dense matrices. The `WignerLaguerre` solver has an optional
-`parallel` parameter which defaults to `true` and uses multithreading to speed up the calculation.
+    wigner(
+        state::QuantumObject{DT,OpType},
+        xvec::AbstractVector,
+        yvec::AbstractVector;
+        g::Real = √2,
+        method::MySolver = WignerClenshaw(),
```

Comment:
maybe we can just put `method:: WignerSolver = WignerClenshaw()` here ?
`MySolver` looks a bit ambiguous.